trade-finance-manager-ui/server/controllers/case/cancellation/validation/validate-bank-request-date.ts

```diff
@@ -0,0 +1,51 @@
+import { applyStandardValidationAndParseDateInput, DayMonthYearInput } from '@ukef/dtfs2-common';
+import { add, isAfter, isBefore, startOfDay } from 'date-fns';
+import { BankRequestDateValidationViewModel } from '../../../../types/view-models';
+
+const DATE_TOO_LATE_MESSAGE = 'The bank request date cannot exceed 12 months in the future from the submission date';
+const DATE_TOO_EARLY_MESSAGE = 'The bank request date cannot exceed 12 months in the past from the submission date';
+
+/**
+ * Returns the error object containing the error summary text with links and the inline error text with refs
+ * @param message the error message to display in the summary and inline
+ * @param refs a list of the specific date fields the error message relates to
+ */
+const getErrorObjectFromMessageAndRefs = (message: string, refs: string[]) => {
+  return {
+    errors: {
+      summary: [{ text: message, href: `#${refs[0]}` }],
+      bankRequestDateError: { message, fields: refs },
+    },
+  };
+};
+
+/**
+ * @param date The entered bank request date
+ * @returns a reason for cancelling errors view model
+ */
+export const validateBankRequestDate = (date: DayMonthYearInput): BankRequestDateValidationViewModel => {
```

Comment:
Do we have any other date validations we could use? I imagine they all have the same rules

Reply:
On line 27 we call `applyStandardValidationAndParseDateInput` which is the commonised date validation checker I made a few weeks back. So this handles all the standard date checks (e.g. missing fields, unexpected values etc.) and this new validator is just to add the specific handling on top of that that is unique to bank request date - which is that it's not more than 12 months in the future or more than 12 months in the past.